app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
```

Comment:
What about this to avoid the confusion:

```
def analytics=(service)
  @analytics = service
end

def analytics
  @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
end
```
  
The first method can them be used for DI use only, and it will behave as expected, allowing multiple different DI within the same request (not that that is needed, but I think you know what I mean).